### PR TITLE
Add profanity profile

### PR DESCRIPTION
**Pull request by adrianlshaw**
My profile for profanity. Comments welcome.

### CODE REVIEW

File: etc/profanity.profile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Firejail profile for profanity
+# Description: profanity is an XMPP-OTR chat client for the terminal
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include unzip.local
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: unzip?

File: etc/profanity.profile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# Persistent global definitions
+include globals.local
+
+ignore net none
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: rm?

File: etc/profanity.profile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+include disable-common.inc
+include disable-devel.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.config/profanity
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: mkdir is only needed before using a whitelist arg

File: etc/profanity.profile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+nou2f
+protocol unix,inet,inet6
+seccomp
+shell none
+
+private-bin profanity
+private-cache
+private-dev
+private-tmp
+private-etc alternatives,localtime,mime.types,resolv.conf,ssl
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: sort etc dev

File: etc/profanity.profile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+memory-deny-write-execute
+noexec ${HOME}
+noexec /tmp
+
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: trailing line

File: etc/profanity.profile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+quiet
+# Persistent local customizations
+include unzip.local
+# Persistent global definitions
+include globals.local
+
+ignore net none
+
+include disable-common.inc
+include disable-devel.inc
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: add disable-exec.inc and disable-interpreters.inc

File: etc/profanity.profile [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.config/profanity
+mkdir ${HOME}/.local/share/profanity
+noblacklist ${HOME}/.config/profanity
+noblacklist ${HOME}/.local/share/profanity
+
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: include whitelist-usr-share-common.inc and whitelist-var-common.inc

File: etc/profanity.profile [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# Firejail profile for profanity
+# Description: profanity is an XMPP-OTR chat client for the terminal
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: profanity has omemo support too :)

File: etc/profanity.profile [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+include disable-xdg.inc
+
+mkdir ${HOME}/.config/profanity
+mkdir ${HOME}/.local/share/profanity
+noblacklist ${HOME}/.config/profanity
+noblacklist ${HOME}/.local/share/profanity
+
+caps.drop all
+netfilter
+nodvd
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: add no3d and nodbus

File: etc/profanity.profile [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+nogroups
+nonewprivs
+noroot
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: add nosound?

File: etc/profanity.profile [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+notv
+nou2f
+protocol unix,inet,inet6
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: add novideo

File: etc/profanity.profile [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+shell none
+
+private-bin profanity
+private-cache
+private-dev
+private-tmp
+private-etc alternatives,localtime,mime.types,resolv.conf,ssl
+
+memory-deny-write-execute
+noexec ${HOME}
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: replaced with disable-exec.inc

**Comment by adrianlshaw** [comment]
Thanks for the review @SkewedZeppelin. Initially this started out as a whitelist, but then I became confused given the structure of existing profiles and the default template. I've addressed all of your comments aside from `disable-interpreters`, because that causes the program to not even start.

Any pro tip on how I can go about effectively debugging the root cause?

**Comment by SkewedZeppelin** [comment]
:+1: 

> not even start

try adding the following above the other noblacklists
```
# Allow python (blacklisted by disable-interpreters.inc)
#include allow-python2.inc
include allow-python3.inc
```

File: etc/profanity.profile [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+include globals.local
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+include whitelist-usr-share-common.inc
```
Comment (SkewedZeppelin) [inline etc/profanity.profile]: nitpick: this should be after the noblacklists and before the general options

**Comment by adrianlshaw** [comment]
Great advice, thank you! I'll be more careful when submitting future profiles.

**Comment by SkewedZeppelin** [comment]
Merged, thanks!

**Pull request merged**